styx-scheduler-service/src/main/java/com/spotify/styx/docker/KubernetesDockerRunner.java

```diff
@@ -246,20 +267,24 @@ void examineRunningWFISandAssociatedPods(PodList podList) {
   @VisibleForTesting
   void pollPods() {
     try {
-      final PodList list = client.pods().list();
-      examineRunningWFISandAssociatedPods(list);
-
-      final int resourceVersion = Integer.parseInt(list.getMetadata().getResourceVersion());
-
-      for (Pod pod : list.getItems()) {
-        logEvent(Watcher.Action.MODIFIED, pod, resourceVersion, true);
-        inspectPod(Watcher.Action.MODIFIED, pod);
-      }
+      tryPollPods();
     } catch (Throwable t) {
       LOG.warn("Error while polling pods", t);
     }
   }
 
+  private synchronized void tryPollPods() {
```

Comment:
As for the lazy creation of `DockerRunner`, this PR changes that behavior to eagerly instantiate the docker runner in `StyxScheduler.restoreState`: https://github.com/spotify/styx/pull/137/files#diff-a3fabd4f5e1e00cf48f40dfdfa3356f1R405

The lazy docker runner instantiation mechanism itself is a bit confusing and should perhaps be changed. Perhaps that could be a separate PR though.